Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -0,0 +1,2978 @@
+                (bond)->atomRefs2[1])];
+        (bond)->atomRefs2[1] = newAtomName;
+      }
+
+      atomPtr->sgroupAttachmentPoint = std::to_string(++attachmentPointsAdded);
+
+      // add an attachentPoint structure
+
+      auto marvinAttachmentPoint = new MarvinAttachmentPoint();
+      marvinSuperatomSgroup->attachmentPoints.push_back(marvinAttachmentPoint);
+      marvinAttachmentPoint->atom = atomPtr->id;
+      marvinAttachmentPoint->bond = bond->id;
+      ;
+      marvinAttachmentPoint->order = std::to_string(attachmentPointsAdded);
+
+      if (coordsExist) {
+        centerOfAttachmentPoints.x += atomPtr->x2;
+        centerOfAttachmentPoints.y += atomPtr->y2;
+      }
+    }
+  }
+
+  // now remove the bonds that were moved to the superGroup from the parent
+
+  for (auto bond : marvinSuperatomSgroup->bonds) {
+    int index = actualParent->getBondIndex(bond->id);
+    actualParent->bonds.erase(actualParent->bonds.begin() + index);
+  }
+
+  if (coordsExist) {
+    if (marvinSuperatomSgroup->attachmentPoints.size() >
+        0)  // Any attachment points?  if so we use the center of the attached
+            // atoms in the supergroup
+    {
+      // put the new dummy atom at the center of the removed group
+
+      dummyParentAtom->x2 = centerOfAttachmentPoints.x /
+                            marvinSuperatomSgroup->attachmentPoints.size();
+      dummyParentAtom->y2 = centerOfAttachmentPoints.y /
+                            marvinSuperatomSgroup->attachmentPoints.size();
+    } else {
+      // No attachments to the supergroup - (probably all atoms in the mol are
+      // in the supergroup) -use the center of all atoms in the group
+      dummyParentAtom->x2 =
+          centerOfGroup.x / marvinSuperatomSgroup->atoms.size();
+      dummyParentAtom->y2 =
+          centerOfGroup.y / marvinSuperatomSgroup->atoms.size();
+    }
+  }
+
+  // move any children from the old expanded group to the new superatomSgroup
+
+  for (auto sgroupPtr : this->sgroups) {
+    moveSgroup(sgroupPtr, marvinSuperatomSgroup, false);
+  }
+  this->sgroups.clear();
+
+  // remove the old expanded superatom sgroup
+
+  parent->sgroups.erase(
+      std::find(parent->sgroups.begin(), parent->sgroups.end(), this));
+  delete this;
+
+  // fix up any siblings that contain all the atoms of this group.
+
+  for (auto sibling : marvinSuperatomSgroup->parent->sgroups) {
+    if (sibling == marvinSuperatomSgroup) {
+      continue;  // not a sibling
+    }
+
+    // see if the new superatom group has all of its atoms in the sibling.  If
+    // to add the dummy atom to the sibling and remove the atoms (there must
+    // be at least one atom in the sibling that is not in the new superatom,
+    // or it would be a child of the superatom already
+
+    auto isSgroupInSetOfAtomsResult =
+        marvinSuperatomSgroup->isSgroupInSetOfAtoms(sibling->atoms);
+
+    switch (isSgroupInSetOfAtomsResult) {
```

Comment:
again, please add a default to the switch

Reply:
the test item is an enum and all enum values are tested in the switch